src/key_manager.rs

```diff
@@ -0,0 +1,113 @@
+use crate::common::Key;
+
+use std::{fmt::Debug, sync::Arc};
+
+/// This is the prefix for all key manager keys.
+pub const KEY_MANAGER_PREFIX: &str = "SSIKeyManager.";
+
+/// This is the default index for a key in the key manager.
+pub const DEFAULT_KEY_INDEX: u8 = 0;
+
+#[uniffi::export(callback_interface)]
+pub trait SecretKeyInterface: Send + Sync {
+    // TODO: Review available methods in android and cryptokit to implement here
+}
+
+#[derive(uniffi::Error, thiserror::Error, Debug)]
+pub enum KeyManagerError {
+    #[error("An unexpected foreign callback error occurred: {0}")]
+    UnexpectedUniFFICallbackError(String),
+    #[error("Failed to generate key")]
+    FailedToGenerateKey,
+    #[error("Failed to encrypt")]
+    FailedToEncrypt,
+    #[error("Failed to decrypt")]
+    FailedToDecrypt,
+    #[error("Failed to sign")]
+    FailedToSign,
+    #[error("Failed to verify")]
+    FailedToVerify,
+    #[error("Failed to reset")]
+    FailedToReset,
+    #[error("Key not found")]
+    KeyNotFound,
+    #[error("Key already exists")]
+    KeyAlreadyExists,
+    #[error("Key invalid")]
+    KeyInvalid,
+}
+
+// Handle unexpected errors when calling a foreign callback
+impl From<uniffi::UnexpectedUniFFICallbackError> for KeyManagerError {
+    fn from(value: uniffi::UnexpectedUniFFICallbackError) -> Self {
+        KeyManagerError::UnexpectedUniFFICallbackError(value.reason)
+    }
+}
+
+/// EncryptedPayload is a struct that holds the IV and ciphertext
+/// of an encrypted payload.
+#[derive(uniffi::Object, Debug, Clone)]
+pub struct EncryptedPayload {
+    iv: Vec<u8>,
+    ciphertext: Vec<u8>,
```

Comment:
TODO: Do we want to expand this type to support key exchange DH encryption? Would include the public key of the intended recipient as a field, possibly optional.

We would also maybe consider expanding the `KeyManager` interface to include encrypting using a recipient's public key. This would be useful for encryption in-transit purposes, versus local encryption at-rest only.

Reply:
Not needed here.